src/component.cpp

```diff
@@ -151,24 +151,21 @@ void Component::removeAllVariables()
     mPimpl->mVariables.clear();
 }
 
-VariablePtr Component::getVariable(size_t index)
+VariablePtr Component::getVariable(size_t index) const
 {
-    return mPimpl->mVariables.at(index);
-}
-
-const VariablePtr& Component::getVariable(size_t index) const
-{
-    return mPimpl->mVariables.at(index);
-}
-
-VariablePtr Component::getVariable(const std::string &name)
-{
-    return mPimpl->mVariables.at(mPimpl->findVariable(name) - mPimpl->mVariables.begin());
+    if (index < mPimpl->mVariables.size()) {
+        return mPimpl->mVariables.at(index);
+    }
+    return nullptr;
 }
 
-const VariablePtr& Component::getVariable(const std::string &name) const
+VariablePtr Component::getVariable(const std::string &name) const
 {
-    return mPimpl->mVariables.at(mPimpl->findVariable(name) - mPimpl->mVariables.begin());
+    auto result = mPimpl->findVariable(name);
+    if (result == mPimpl->mVariables.end()) {
+        return nullptr;
+    }
+    return *result;
```

Comment:
I will leave this for when #140 is addressed because all this code is related and will make it consistent throughout all objects then.  Just wanting to avoid dealing with merge conflicts.